### PR TITLE
[Enhancement] Improve only_null check

### PR DESCRIPTION
**Pull request by Seaven**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Review by murphyatwork — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/column/const_column.h [hunk 1]
```diff
@@ -50,7 +50,10 @@ class ConstColumn final : public ColumnFactory<Column, ConstColumn> {
 
     bool is_null(size_t index) const override { return _data->is_null(0); }
 
-    bool only_null() const override { return _data->is_null(0); }
+    bool only_null() const override {
+        DCHECK(_data->is_nullable() ? _data->is_null(0) : true);
```
Comment (satanson) [inline be/src/column/const_column.h:54]: typo?
```
DCHECK(_data->is_nullable() ? _data->is_null(0) : true);
```

```
DCHECK(_data->is_nullable() ? _data->is_null(0) : false);
```

**Review by stdpain — APPROVED** [review]

File: be/src/column/const_column.h (hunk 1 above)
Comment (Seaven) [reply be/src/column/const_column.h:54]: nonono, is right if not nullable

**Review by satanson — APPROVED** [review]

**Pull request merged**